addons/mail/static/src/core/common/composer_state_service.js

```diff
@@ -0,0 +1,33 @@
+import { reactive } from "@odoo/owl";
+import { registry } from "@web/core/registry";
+
+export const composerStateService = {
+    dependencies: ["mail.store", "legacy_multi_tab"],
+    /**
+     * @param {import("@web/env").OdooEnv}
+     * @param {Partial<import("services").Services>} services
+     */
+    start(env, { legacy_multi_tab }) {
+        const state = reactive({
+            htmlEnabled: legacy_multi_tab.getSharedValue("mail.html_composer.enabled", false),
+            setHtmlComposer() {
+                state.htmlEnabled = true;
+                legacy_multi_tab.setSharedValue("mail.html_composer.enabled", true);
+            },
+            setTextareaComposer() {
+                state.htmlEnabled = false;
+                legacy_multi_tab.setSharedValue("mail.html_composer.enabled", false);
+            },
+        });
+
+        legacy_multi_tab.bus.addEventListener("shared_value_updated", ({ detail }) => {
+            if (detail.key === "mail.html_composer.enabled") {
+                state.htmlEnabled = JSON.parse(detail.newValue);
+            }
+        });
+
+        return state;
+    },
+};
+
+registry.category("services").add("mail.composer_state_service", composerStateService);
```

Comment:
We don't usually suffix services with the _service keyword.

Reply:
I'd also just name `composer_service` / `mail.composer`, simpler name helps readability and keeps overall concepts in discuss code simple as it should be.